shell/machine-config/harvester.vue

```diff
@@ -537,7 +538,15 @@ export default {
         <h3 class="mt-20">
           {{ t("workload.container.titles.podScheduling") }}
         </h3>
-        <PodAffinity :mode="mode" :value="vmAffinity" :nodes="allNodeObjects" :has-nodes-and-ns="isImportCluster" @update="updateScheduling" />
+        <PodAffinity
+          :mode="mode"
+          :value="vmAffinity"
+          :nodes="allNodeObjects"
+          :has-nodes="allNodeObjects.length > 0"
+          :has-namespaces="isImportCluster"
+          override-store="harvester"
```

Comment:
Since the harvester machine-config component already has a list of `namespaceOptions`, we can pass that through as a prop and override `choices` in `allNamespaces`. Then we keep harvester-specific code out of the PodAffinity component.